libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -468,28 +468,28 @@ proc rebalanceMesh(g: GossipSub, topic: string) {.async.} =
         x.peerId notin g.backingOff
 
       # shuffle anyway, score might be not used
-      shuffle(grafts)
+      shuffle(candidates)
 
       # sort peers by score, high score first, we are grafting
-      grafts.sort(byScore, SortOrder.Descending)
+      candidates.sort(byScore, SortOrder.Descending)
 
       # Graft peers so we reach a count of D
-      grafts.setLen(min(grafts.len, g.parameters.dOut))
+      candidates.setLen(min(candidates.len, g.parameters.dOut))
 
-      trace "grafting outbound peers", topic, peers = grafts.len
+      trace "grafting outbound peers", topic, peers = candidates.len
 
-      for peer in grafts:
+      for peer in candidates:
         if g.mesh.addPeer(topic, peer):
           g.grafted(peer, topic)
           g.fanout.removePeer(topic, peer)
-          grafting &= peer
+          grafts &= peer
 
 
   if g.mesh.peers(topic) > g.parameters.dHigh:
     # prune peers if we've gone over Dhi
     prunes = toSeq(g.mesh[topic])
     # avoid pruning peers we are currently grafting in this heartbeat
-    prunes.keepIf do (x: PubSubPeer) -> bool: x notin grafting
+    prunes.keepIf do (x: PubSubPeer) -> bool: x notin grafts
     let mesh = prunes
```

Comment:
```suggestion
    let meshOutbound = prunes.countIt(it.outbound)
```

also, shouldn't this happen before the `grafts` filter?

Reply:
not sure I get this one, good suggestion about `countIt` but the grafts thing is to avoid considering currently `grafting` peers